### DISCUSSION TITLE
wavesurfer appeared to be freezing, possible solution

### DISCUSSION THREAD

**Discussion by edwardsmarkf** [katspaugh/wavesurfer.js#2887] [General]
hello - this has probably been addressed, but i was not sure how to search for it.

my latest wavesurfer site with 7.x seemed to be freezing up the Chrome (latest) browser by doing a `.load()` two or three times without hitting the stop button first.   here is a possibly very crude way to deal with it

```
        wavesurfer.stop();
        setTimeout( () => { wavesurfer.empty()                  }       , 200   )       ;
        setTimeout( () => ( wavesurfer.seekTo(0)                )       , 400   )       ;       // do we really need this ????
        setTimeout( () => { wavesurfer.load(audioFile)      }       , 600   )       ;
```
it appears that entering a slight delay fixed it for me.

thoughts or suggestions?   (suggestions other than changing careers to something non-tech please)

**Comment by katspaugh** [comment]
Sounds like a bug! I guess we should call stop internally before loading a new file.

**Reply by edwardsmarkf** [reply]
actually i was thinking something like turn stop into a promise ?

```
wavesurfer.stop().then( () => {
    wavesurfer.load(audioFile);
  }).catch( (err) => {
      console.error(err);
  });
  ```

**Reply by katspaugh** [reply]
I've created an issue to track this bug (#2958). Thanks!
